src/library/compiler/ir_interpreter.cpp

```diff
@@ -787,25 +793,30 @@ class interpreter {
 
     /** \brief Return cached lookup result for given unmangled function name in the current binary. */
     symbol_cache_entry lookup_symbol(name const & fn) {
-        if (symbol_cache_entry const * e = m_symbol_cache.find(fn)) {
+        std::shared_lock<std::shared_timed_mutex> lock(*g_symbol_cache_mutex);
```

Comment:
Do we actually want a `shared_timed_mutex` or a `shared_mutex` (from C++17, since IIRC the old `shared_mutex` was renamed to `shared_timed_mutex` in C++14)? I don't think we have any need for the timeout.

Reply:
We're currently using C++14, I didn't want to bump that without further need